src/store/modules/layers.store.js

```diff
@@ -5,8 +5,14 @@ import { getGpxExtent } from '@/utils/gpxUtils.js'
 import { getKmlExtent, parseKmlName } from '@/utils/kmlUtils'
 import log from '@/utils/logging'
 
-const getActiveLayersById = (state, layerId) =>
-    state.activeLayers.filter((layer) => layer.id === layerId)
+const getActiveLayersById = (state, layerId, isExternal = null, baseUrl = null) => {
+    return state.activeLayers.filter((layer) => {
+        const matchesLayerId = layer.id === layerId
+        const matchesIsExternal = isExternal === null || layer.isExternal === isExternal
+        const matchesBaseUrl = baseUrl === null || layer.baseUrl === baseUrl
+        return matchesLayerId && matchesIsExternal && matchesBaseUrl
```

Comment:
this bit of code is repeated three times across the store. It would be cleaner to turn it into a function like so : 
``` 
function matchTwoLayers(layerId, isExternal=null, baseUrl=null, layerToMatch) {
const matchesLayerId = layer.id === layerId
        const matchesIsExternal = isExternal === null || layer.isExternal === isExternal
        const matchesBaseUrl = baseUrl === null || layer.baseUrl === baseUrl
        return matchesLayerId && matchesIsExternal && matchesBaseUrl
}
```

then you call it here (line 9), in getActiveLayersById (line 178) and in removeLayerById (line 818) by checking for a false value.That way, if we need to change the way we filter layers by their various identifiers, we can do it by changing the code only in one place.